File: 12 Last Stone Weight/Last Stone Weight.cpp
class Solution
{
public:
    int lastStoneWeight(vector<int> &stones)
    {
        sort(stones.begin(), stones.end());
        while (1)
        {
            if (stones.size() < 2)
            {
                break;
            }
            int diff = stones[stones.size() - 1] - stones[stones.size() - 2];
            stones.pop_back();
            stones.pop_back();
            if (diff != 0)
            {
                stones.push_back(diff);
            }
            sort(stones.begin(), stones.end());
        }
        if (stones.size() == 1)
            return stones[0];
        else
            return 0;
    }
};

File: 12 Last Stone Weight/Last Stone Weight.js
/**
 * @param {number[]} stones
 * @return {number}
 */
var lastStoneWeight = function (stones) {
    stones.sort((a, b) => a - b);
    while (1) {
        if (stones.length < 2)
            break;
        let diff = stones[stones.length - 1] - stones[stones.length - 2];
        stones.pop();
        stones.pop();
        if (diff != 0)
            stones.push(diff);
        stones.sort((a, b) => a - b);
    }
    if (stones.length == 1)
        return stones[0];
    else
        return 0;

};